lib/manageiq/messaging/kafka/common.rb

```diff
@@ -53,20 +53,23 @@ def queue_for_publish(options)
           body, kafka_opts = for_publish(options)
           kafka_opts[:headers][:message_type] = options[:message] if options[:message]
           kafka_opts[:headers][:class_name] = options[:class_name] if options[:class_name]
+          kafka_opts[:headers].merge!(options[:headers].except(*message_header_keys)) if options.key?(:headers)
```

Comment:
do we silently drop the conflicted header keys or shall we raise an error (here and everywhere)?

Reply:
This will just strip out any internal keys, can we push raising an exception to a follow-up? I have a dependent PR that other teams have been waiting on for a while